network-policy.go

```diff
@@ -0,0 +1,201 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocp
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/kube-burner/kube-burner/pkg/config"
+	kutil "github.com/kube-burner/kube-burner/pkg/util"
+	"github.com/kube-burner/kube-burner/pkg/workloads"
+	routev1 "github.com/openshift/api/route/v1"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/utils/ptr"
+
+	openshiftrouteclientset "github.com/openshift/client-go/route/clientset/versioned"
+)
+
+var networkPolicyProxyPort int32 = 9002
+var networkPolicyProxy = "network-policy-proxy"
+var networkPolicyProxyLabel = map[string]string{"app": networkPolicyProxy}
+var networkPolicyProxyRouteName string
+
+var networkPolicyProxyPod = &corev1.Pod{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      networkPolicyProxy,
+		Namespace: networkPolicyProxy,
+		Labels:    networkPolicyProxyLabel,
+	},
+	Spec: corev1.PodSpec{
+		TerminationGracePeriodSeconds: ptr.To[int64](0),
+		Containers: []corev1.Container{
+			{
+				Image:           "quay.io/cloud-bulldozer/netpolproxy:latest",
+				Name:            networkPolicyProxy,
+				ImagePullPolicy: corev1.PullAlways,
+				SecurityContext: &corev1.SecurityContext{
+					AllowPrivilegeEscalation: ptr.To[bool](false),
+					Capabilities:             &corev1.Capabilities{Drop: []corev1.Capability{"ALL"}},
+					RunAsNonRoot:             ptr.To[bool](true),
+					SeccompProfile:           &corev1.SeccompProfile{Type: corev1.SeccompProfileTypeRuntimeDefault},
+					RunAsUser:                ptr.To[int64](1000),
+				},
+			},
+		},
+	},
+}
+
+var networkPolicyProxySvc = &corev1.Service{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:      networkPolicyProxy,
+		Namespace: networkPolicyProxy,
+		Labels:    networkPolicyProxyLabel,
+	},
+	Spec: corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
+			{
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.Parse(fmt.Sprintf("%d", networkPolicyProxyPort)),
+				Port:       80,
+				Name:       "http",
+			},
+		},
+		Type:     corev1.ServiceType("ClusterIP"),
+		Selector: networkPolicyProxyLabel,
+	},
+}
+
+var networkPolicyProxyRoute = routev1.Route{
+	ObjectMeta: metav1.ObjectMeta{
+		Name:   networkPolicyProxy,
+		Labels: networkPolicyProxyLabel,
+	},
+	Spec: routev1.RouteSpec{
+		Port: &routev1.RoutePort{TargetPort: intstr.FromString("http")},
+		To: routev1.RouteTargetReference{
+			Name: networkPolicyProxy,
+		},
+	},
+}
+
+// create proxy pod with route
+func deployAssets(uuid string, clientSet kubernetes.Interface, restConfig *rest.Config) error {
```

Comment:
Can we move all these assets to the config as well? If we intent to run them at start only once we can do that by either of the options below
* [Add them to start of config](https://github.com/kube-burner/kube-burner-ocp/blob/main/cmd/config/networkpolicy-matchlabels/networkpolicy-matchlabels.yml#L26-L34)

   OR

* [By using run once option we have](https://kube-burner.github.io/kube-burner/latest/reference/configuration/?h=runonce#runonce)

Reply:
Thanks Vishnu. Initially I was doing the same 
https://github.com/venkataanil/files/blob/master/kube-burrner-netpol/network-policy.yml#L11 
However, the route will be used inside core kube-burner measurement code, so earlier I was manually adding it in the core kube-burner measurement code
https://github.com/venkataanil/files/blob/master/kube-burrner-netpol/netpol_latency.go#L65

Now kube-burner-ocp creates it and passes it as environment variable https://github.com/kube-burner/kube-burner-ocp/pull/117/files#diff-43d885d93cbe44660401e7017fb54c6cfe09507946b9374ad7d7b0a0485e5143R172 
This env variable is consumed in the template
https://github.com/kube-burner/kube-burner-ocp/pull/117/files#diff-22fffda545f823bb596360df5ac5fffdbd6b233b9b662a67ddd506d48162d677R11
Finally core kube-burner measurement code reads from the config file https://github.com/kube-burner/kube-burner/pull/679/files#diff-a6045e2aa0d1b86a2c266acf164b4102c828a73863f6212d7eff79a34be53e04R496 

I was following similar approach of service_latency.go https://github.com/kube-burner/kube-burner/blob/main/pkg/measurements/service_latency.go#L73 
However I couldn't directly define deploy_assets() which creates this proxy route in 
 pkg/measurements/netpol_latency.go (https://github.com/kube-burner/kube-burner/pull/679/files#diff-a6045e2aa0d1b86a2c266acf164b4102c828a73863f6212d7eff79a34be53e04) as "route" resource belongs to ocp and not k8s. So I needed to move it to kube-burner-ocp.

I can move it to template if there is a way measurement code pkg/measurements/netplo_latency.go get this route information.

Reply:
Thanks for the detailed explanation. My concern was to just move these assets creation to the top of the workload config file in a desired order but not to kube-burner core repo.

Reply:
This is moved now.